esinstall/src/entrypoints.ts

```diff
@@ -0,0 +1,337 @@
+    return path.join(depManifestLoc, '..', typesLoc);
+  }
+  // Sometimes packages don't give an entrypoint, assuming you'll fall back to "index.js".
+  if (!foundEntrypoint) {
+    foundEntrypoint = 'index.js';
+  }
+  if (typeof foundEntrypoint !== 'string') {
+    throw new Error(`"${dep}" has unexpected entrypoint: ${JSON.stringify(foundEntrypoint)}.`);
+  }
+
+  return realpathSync.native(
+    require.resolve(path.join(depManifestLoc || '', '..', foundEntrypoint)),
+  );
+}
+
+const picoMatchGlobalOptions = Object.freeze({
+  capture: true,
+  noglobstar: true,
+});
+
+function* forEachExportEntry(
+  exportField: ExportField,
+): Generator<[string, unknown], any, undefined> {
+  const simpleExportMap = findExportMapEntry(exportField);
+
+  // Handle case where export map is a string, or if there‘s only one file in the entire export map
+  if (simpleExportMap) {
+    yield ['.', simpleExportMap];
+
+    return;
+  }
+
+  for (const [key, val] of Object.entries(exportField)) {
+    // skip invalid entries
+    if (!key.startsWith('.')) {
+      continue;
+    }
+
+    yield [key, val];
+  }
+}
+
+function* forEachWildcardEntry(
+  key: string,
+  value: string,
+  cwd: string,
+): Generator<[string, string], any, undefined> {
+  // Creates a regex from a pattern like ./src/extras/*
+  let expr = pm.makeRe(value, picoMatchGlobalOptions);
+
+  // The directory, ie ./src/extras
+  let valueDirectoryName = path.dirname(value);
+  let valueDirectoryFullPath = path.join(cwd, valueDirectoryName);
+
+  if (existsSync(valueDirectoryFullPath)) {
+    let filesInDirectory = readdirSync(valueDirectoryFullPath);
+
+    for (let filename of filesInDirectory) {
+      // Create a relative path for this file to match against the regex
+      // ex, ./src/extras/one.js
+      let relativeFilePath = path.join(valueDirectoryName, filename);
+      let match = expr.exec(relativeFilePath);
+
+      if (match && match[1]) {
+        let [matchingPath, matchGroup] = match;
+        let normalizedKey = key.replace('*', matchGroup);
+
+        // Normalized to posix paths, like ./src/extras/one.js
+        let normalizedFilePath =
+          '.' + path.posix.sep + matchingPath.split(path.sep).join(path.posix.sep);
+
+        // Yield out a non-wildcard match, for ex.
+        // ['./src/extras/one', './src/extras/one.js']
+
+        yield [normalizedKey, normalizedFilePath];
+      }
+    }
+  }
+}
+
+function* forEachExportEntryExploded(
+  exportField: ExportField,
+  cwd: string,
+): Generator<[string, unknown], any, undefined> {
+  for (const [key, val] of forEachExportEntry(exportField)) {
+    // Deprecated but we still want to support this.
+    // https://nodejs.org/api/packages.html#packages_subpath_folder_mappings
+    if (key.endsWith('/')) {
+      if (typeof val !== 'string') {
+        continue;
+      }
+
+      // There isn't a clear use-case for this, so we are assuming it's not needed for now.
```

Comment:
👍🏻

Reply:
This is based on a conversation with @FredKSchott where Preact has a mapping of `"./": "./"` that would we need to add support for. Since this isn't a use-case for this in any other package we talked about skipping this case for now.